Lib/pty.py

```diff
@@ -132,14 +134,22 @@ def _copy(master_fd, master_read=_read, stdin_read=_read):
             pty master -> standard output   (master_read)
             standard input -> pty master    (stdin_read)"""
     fds = [master_fd, STDIN_FILENO]
-    while True:
-        rfds, wfds, xfds = select(fds, [], [])
+    while fds:
+        rfds, _wfds, _xfds = select(fds, [], [])
+
         if master_fd in rfds:
-            data = master_read(master_fd)
+            # Some OSes signal EOF by returning an empty byte string,
+            # some throw OSErrors.
+            try:
+                data = master_read(master_fd)
+            except OSError:
+                data = b""
             if not data:  # Reached EOF.
-                fds.remove(master_fd)
+                return    # Assume the child process has exited and is
```

Comment:
mmm I would return something i.e. False

Reply:
Why would this return `False`? And what condition ought to return `True`?  I don't think there is any basis to say that one execution path was "successful" and others were not.  There's nothing inherently successful about returning because only the `master_fd` signallled EOF versus the case that both `master_fd` _and_ `stdin` received an EOF on the same loop.

In my mind, the only way to unsuccessfully return from this method would be for an unexpected error to be thrown, and an `OSError` is _expected_ to be thrown by reading from a closed `fd` on certain platforms.

Reply:
Just to make sure that we are on the same page: I believe that in this case, the OSError that is expected on linux when attempting to read from master is a result of the slave end of the pty closing. It would also happen if the master is closed, but that is not what is happening here since the master remains open until it is closed toward the end of spawn().